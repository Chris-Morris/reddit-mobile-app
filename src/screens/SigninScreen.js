import React, { useState, useContext, useEffect, createContext } from 'react';
import { View, Text, TextInput, TouchableOpacity, StyleSheet } from 'react-native';

import authContext from '../../context/authContext';
const AuthContext = createContext(authContext);

const SigninScreen = () => {
    const [username, setUsername] = useState('');
    const [invalidUsername, setInvalidUsername] = useState(false);
    const [password, setPassword] = useState('');
    const [invalidPassword, setInvalidPassword] = useState(false);
    const [showButton, setShowButton] = useState(false);

    const { signIn } = useContext(AuthContext);

    useEffect(() => {
        if (username.length < 5) {
            setInvalidUsername(true);
        } else {
            setInvalidUsername(false);
        };

        if (username.length === 0) {
            setInvalidUsername(false);
        };

        if (password.length < 5) {
            setInvalidPassword(true);
        } else {
            setInvalidPassword(false);
        };

        if (password.length === 0) {
            setInvalidPassword(false);
        };

        if (username.length >= 5 && password.length >= 5) {
            setInvalidUsername(false);
            setInvalidPassword(false);
            setShowButton(true);
        };
    }, [username, password]);

    return (
        <View style={signinStyle.container} >


            {!invalidUsername ?
                <View>
                    <Text>Please enter your username</Text>
                    <Text>(min 5 characters)</Text>
                    <TextInput
                        autoCapitalize='none'
                        autoCorrect={false}
                        clearButtonMode='always'
                        style={signinStyle.input}
                        placeholder="Username"
                        value={username}
                        onChangeText={setUsername}
                    />
                </View>
                :
                <View>
                    <Text>Please enter a valid username</Text>
                    <Text>(min 5 characters)</Text>
                    <TextInput
                        autoCapitalize='none'
                        autoCorrect={false}
                        clearButtonMode='always'
                        style={signinStyle.invalidInput}
                        placeholder="Username"
                        value={username}
                        onChangeText={setUsername}
                    />
                </View>
            }

            {invalidPassword ?
                <View>
                    <Text>Please enter a valid password</Text>
                    <TextInput
                        autoCapitalize='none'
                        autoCorrect={false}
                        clearButtonMode='always'
                        style={signinStyle.invalidInput}
                        placeholder="Password"
                        value={password}
                        onChangeText={setPassword}
                    />
                </View>
                :
                <View>
                    <Text>Please enter your password</Text>
                    <TextInput
                        autoCapitalize='none'
                        autoCorrect={false}
                        clearButtonMode='always'
                        style={signinStyle.input}
                        placeholder="Password"
                        value={password}
                        onChangeText={setPassword}
                    />
                </View>
            }
            {showButton ?
                <TouchableOpacity style={signinStyle.button} onPress={() => signIn({ username, password })}><Text style={{ color: 'white' }} >Sign In</Text></TouchableOpacity>
                :
                <TouchableOpacity style={signinStyle.inactiveButton} ><Text style={{ color: 'white' }} >Sign In</Text></TouchableOpacity>
            }
        </View >
    );
}

const signinStyle = StyleSheet.create({
    container: {
        height: 300, marginTop: 50, justifyContent: 'space-around', alignItems: 'center'
    },
    input: {
        height: 30, width: 200, backgroundColor: 'rgb(236, 240, 246)', borderStyle: 'solid', borderColor: 'rgb(0, 122, 204)', borderWidth: 1, borderRadius: 5, paddingLeft: 10
    },
    invalidInput: {
        height: 30, width: 200, backgroundColor: 'rgb(236, 240, 246)', borderStyle: 'solid', borderColor: 'rgb(255, 2, 2)', borderWidth: 1, borderRadius: 5, paddingLeft: 10
    },
    button: {
        backgroundColor: 'rgb(0, 122, 204)', height: 30, width: 124, borderRadius: 5, justifyContent: 'center', alignItems: 'center'
    },
    inactiveButton: {
        backgroundColor: 'rgb(51, 51, 51)', height: 30, width: 124, borderRadius: 5, justifyContent: 'center', alignItems: 'center'
    }
});

export default SigninScreen
