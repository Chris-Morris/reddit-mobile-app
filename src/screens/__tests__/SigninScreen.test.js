import React from 'react';
import { render, cleanup, screen } from '@testing-library/react-native';

import SigninScreen from '../SigninScreen';

describe('Sign In Screen', () => {
    afterEach(cleanup);

    // Mobile version
    test('renders username input', () => {
        // Render the Sign In screen
        const { getByPlaceholderText } = render(<SigninScreen />);

        // Extract the Username input
        const username = getByPlaceholderText('Username');

        // Assert the damn thing exists
        expect(username).toBeInTheDocument();
    });

    // Desktop version
    test('renders username input', () => {
        // Render the Sign In screen
        render(<SigninScreen />);

        // Extract the Username input
        const username = screen.getByPlaceholderText('Username');

        // Assert the damn thing exists
        expect(username).toBeInTheDocument();
    });
});