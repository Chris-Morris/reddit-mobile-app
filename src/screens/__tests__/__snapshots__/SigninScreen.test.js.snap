// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Screen renders username input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 300,
      "justifyContent": "space-around",
      "marginTop": 50,
    }
  }
>
  <View>
    <Text>
      Please enter your username
    </Text>
    <Text>
      (min 5 characters)
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Username"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "rgb(236, 240, 246)",
          "borderColor": "rgb(0, 122, 204)",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 30,
          "paddingLeft": 10,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View>
    <Text>
      Please enter your password
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "rgb(236, 240, 246)",
          "borderColor": "rgb(0, 122, 204)",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 30,
          "paddingLeft": 10,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(51, 51, 51)",
        "borderRadius": 5,
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "width": 124,
      }
    }
    testID="button"
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Sign In
    </Text>
  </View>
</View>
`;
